processor/src/main/java/org/mapstruct/ap/internal/model/source/BeanMapping.java

```diff
@@ -61,12 +63,25 @@ public static BeanMapping fromPrism(BeanMappingPrism beanMapping, ExecutableElem
             beanMapping.qualifiedByName(),
             resultTypeIsDefined ? beanMapping.resultType() : null );
 
-        return new BeanMapping(cmp, nullValueMappingStrategy );
+        //TODO Do we want to add the reporting policy to the BeanMapping as well? To give more granular support?
```

Comment:
Can you give an example where it would be useful? Probably best to discuss separately, though. If you like create an issue for the discussion's sake, we can close it at the end if we agree it's not needed.